lib/irb/cmd/history.rb

```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+
+require "stringio"
+require_relative "nop"
+require_relative "../pager"
+
+module IRB
+  # :stopdoc:
+
+  module ExtendCommand
+    class History < Nop
+      category "IRB"
+      description "Shows the input history. `-g [query]` or `-G [query]` allows you to filter the output."
+
+      def self.transform_args(args)
+        if match = args&.match(/\A(?<args>.+\s|)(-g|-G)\s+(?<grep>[^\s]+)\s*\n\z/)
+          args = match[:args]
+          "#{args}#{',' unless args.chomp.empty?} grep: /#{match[:grep]}/"
+        else
+          args
+        end
+      end
+
+      def execute(*arg, grep: nil)
+        formatted_inputs = irb_context.io.class::HISTORY.each_with_index.reverse_each.map do |input, index|
+          header = "#{index}: "
+
+          first_line, *other_lines = input.split("\n") || [""]
```

Comment:
The `|| [""]` has no effect because `input.split("\n")` will always return an array, (not nil or false)
and if input is `""`, split will return `[]` and `first_line.prepend header` will raise NoMethodError.

One simple idea is
```ruby
first_line = "#{header}#{first_line}"
```
It works even if input is `""` and first_line is `nil`

Reply:
Thank you for such a thorough review!

https://github.com/ruby/irb/pull/761/commits/5b64a5be9abe985161aebe92b37af14d4f45b9c6